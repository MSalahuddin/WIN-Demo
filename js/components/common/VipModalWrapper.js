import React, { useState } from 'react';
import Modal from 'react-native-modal';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';

import Text, { SIZE } from './Text';
import Button, { BUTTON_COLOR_SCHEME } from './Button';
import Banner, { BANNER_TYPE } from './Banner';
import { scaleHeight, getWindowWidth, scaleWidth, scale, heightRatio } from '../../platformUtils';
import { color } from '../../styles';
import { accountProfileStrings } from '../../stringConstants';
import { medalStar, chickenWithBalloon } from '../../../assets/images';

const StyledModal = styled(Modal)`
  background-color: ${color.popupBlack};
  justify-content: center;
  margin: 0;
`;

const InnerModalContainer = styled.View`
  align-items: center;
`;

const StarIcon = styled.Image`
  height: ${scaleHeight(88)};
  margin-bottom: ${scaleHeight(24)};
  width: ${scaleHeight(88)};
`;

const mascotPositionVariation = heightRatio < 1 ? scaleHeight(75) : scaleHeight(52);
const ChickenImage = styled.Image`
  position: absolute;
  right: 0;
  top: ${({ top }) => scaleHeight(top) - mascotPositionVariation};
`;

const Backdrop = styled.View`
  align-items: center;
  background-color: ${color.white};
  border-bottom-left-radius: ${scaleHeight(8)};
  border-bottom-right-radius: ${scaleHeight(8)};
  height: ${({ height }) => scaleHeight(height)};
  justify-content: flex-start;
  margin-top: ${-scaleHeight(20)};
  padding-top: ${({ paddingTop }) => scaleHeight(paddingTop)};
  padding-horizontal: ${scale(28)};
  shadow-color: ${color.blackShadow};
  shadow-offset: 4px 4px;
  shadow-opacity: ${1};
  shadow-radius: 2px;
  width: 85%;
  z-index: -1;
`;

const ButtonWrapper = styled(Button)`
  margin-top: ${({ marginTop }) => scaleHeight(marginTop)};
`;

const ButtonContentContainer = styled.View`
  align-items: center;
  flex-direction: row;
  justify-content: center;
`;

const ButtonText = styled(Text)`
  margin-top: ${scaleHeight(7)};
`;

const VipModalWrapper = ({ isVisible, setVisible, children, showChicken, backdropHeight, backdropTopPadding }) => {
  const [bannerPosition, setBannerPosition] = useState(0);

  return (
    <StyledModal isVisible={isVisible} hideModalContentWhileAnimating>
      <InnerModalContainer>
        <StarIcon source={medalStar} resizeMode="contain" />
        <Banner
          bannerType={BANNER_TYPE.STAR}
          label={accountProfileStrings.vip}
          textSize={SIZE.BANNER_LARGE}
          width={getWindowWidth() - scaleWidth(32)}
        />
        <Backdrop
          height={backdropHeight}
          paddingTop={backdropTopPadding}
          onLayout={event => {
            const { y } = event.nativeEvent.layout;
            setBannerPosition(y);
          }}
        >
          {children}
        </Backdrop>
        <ButtonWrapper
          testID="dismiss-button"
          borderRadius={44}
          height={88}
          width={271}
          theme={BUTTON_COLOR_SCHEME.GREEN}
          onPress={setVisible}
          marginTop={-46}
        >
          <ButtonContentContainer>
            <ButtonText color={color.white} size={SIZE.XLARGE}>
              {accountProfileStrings.awesome}
            </ButtonText>
          </ButtonContentContainer>
        </ButtonWrapper>
      </InnerModalContainer>
      {showChicken && <ChickenImage source={chickenWithBalloon} resizeMode="contain" top={bannerPosition} />}
    </StyledModal>
  );
};

VipModalWrapper.defaultProps = {
  showChicken: false,
  backdropHeight: 177,
  backdropTopPadding: 40
};

VipModalWrapper.propTypes = {
  children: PropTypes.node.isRequired,
  isVisible: PropTypes.bool.isRequired,
  setVisible: PropTypes.func.isRequired,
  showChicken: PropTypes.bool,
  backdropHeight: PropTypes.number,
  backdropTopPadding: PropTypes.number
};

export default VipModalWrapper;
