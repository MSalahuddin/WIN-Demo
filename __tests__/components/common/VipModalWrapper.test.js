import React from 'react';
import { View } from 'react-native';
import { render, fireEvent } from 'react-native-testing-library';

import VipModalWrapper from '../../../js/components/common/VipModalWrapper';
import MockProvider from '../../MockProvider';

describe('Component VipModalWrapper', () => {
  const setVisible = jest.fn();

  test('renders correctly', () => {
    const output = render(
      <MockProvider>
        <VipModalWrapper isVisible setVisible={setVisible}>
          <View />
        </VipModalWrapper>
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('renders correctly when !isVisible', () => {
    const output = render(
      <MockProvider>
        <VipModalWrapper isVisible={false} setVisible={setVisible}>
          <View />
        </VipModalWrapper>
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('renders correctly when showChicken', () => {
    const output = render(
      <MockProvider>
        <VipModalWrapper isVisible setVisible={setVisible} showChicken>
          <View />
        </VipModalWrapper>
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('calls onPress when button is pressed', () => {
    const { getByTestId } = render(
      <MockProvider>
        <VipModalWrapper isVisible setVisible={setVisible}>
          <View />
        </VipModalWrapper>
      </MockProvider>
    );
    fireEvent(getByTestId('dismiss-button'), 'press');
    expect(setVisible).toHaveBeenCalled();
  });
});
