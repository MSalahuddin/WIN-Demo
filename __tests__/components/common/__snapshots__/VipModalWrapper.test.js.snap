// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component VipModalWrapper renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={true}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={true}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "backgroundColor": "rgba(34,34,34, 0.6)",
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Array [
            Object {
              "height": 176,
              "marginBottom": 48,
              "width": 176,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
        width={686}
      >
        <Image
          bannerType="STAR"
          resizeMode="contain"
          source={1}
          style={
            Array [
              Object {
                "height": 125.44,
                "width": 686,
              },
            ]
          }
          width={686}
        />
        <Text
          alignCenter={false}
          allowFontScaling={false}
          bannerType="STAR"
          color="#000000"
          fontFamily="BOLD"
          fontSize={22}
          isUnderlined={false}
          isUppercase={true}
          maxFontSizeMultiplier={1.3}
          size="BANNER_LARGE"
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "CalibreBold",
                "fontSize": 34.5,
                "justifyContent": "center",
                "textAlign": "left",
                "textDecorationColor": "#000000",
                "textDecorationLine": "none",
                "textDecorationStyle": "solid",
                "textTransform": "uppercase",
              },
              Object {
                "color": "#FFFFFF",
                "letterSpacing": 1.5,
                "position": "absolute",
              },
            ]
          }
        >
          VIP
        </Text>
      </View>
      <View
        height={177}
        onLayout={[Function]}
        paddingTop={40}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderBottomLeftRadius": 16,
              "borderBottomRightRadius": 16,
              "height": 354,
              "justifyContent": "flex-start",
              "marginTop": -40,
              "paddingHorizontal": 42,
              "paddingTop": 80,
              "shadowColor": "rgba(53, 53, 53, 0.3)",
              "shadowOffset": Object {
                "height": 4,
                "width": 4,
              },
              "shadowOpacity": 1,
              "shadowRadius": 2,
              "width": "85%",
              "zIndex": -1,
            },
          ]
        }
      >
        <View />
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginHorizontal": 10,
              "marginVertical": 10,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(96,202,142,1)",
              "borderRadius": 88,
              "height": 176,
              "justifyContent": "center",
              "marginTop": -92,
              "opacity": 1,
              "width": 542,
            }
          }
          testID="dismiss-button"
        >
          <View
            borderRadius={44}
            colorScheme="rgba(128,229,173,1)"
            height={88}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(128,229,173,1)",
                  "borderRadius": 88,
                  "height": 166,
                  "justifyContent": "center",
                  "width": 536,
                },
              ]
            }
            width={271}
          >
            <View
              borderRadius={44}
              colorScheme="rgba(114,213,158,1)"
              height={88}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "rgba(114,213,158,1)",
                    "borderRadius": 108,
                    "height": 162,
                    "justifyContent": "center",
                    "marginTop": 8,
                    "width": 536,
                  },
                ]
              }
              width={271}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Text
                  alignCenter={false}
                  allowFontScaling={true}
                  color="#FFFFFF"
                  fontFamily="MEDIUM"
                  isUnderlined={false}
                  isUppercase={false}
                  maxFontSizeMultiplier={1.3}
                  size="XLARGE"
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontFamily": "Calibre-Medium",
                        "fontSize": 36,
                        "justifyContent": "center",
                        "textAlign": "left",
                        "textDecorationColor": "#FFFFFF",
                        "textDecorationLine": "none",
                        "textDecorationStyle": "solid",
                        "textTransform": "none",
                      },
                      Object {
                        "marginTop": 14,
                      },
                    ]
                  }
                >
                  Awesome!
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`Component VipModalWrapper renders correctly when !isVisible 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={true}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={false}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={true}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "backgroundColor": "rgba(34,34,34, 0.6)",
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Array [
            Object {
              "height": 176,
              "marginBottom": 48,
              "width": 176,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
        width={686}
      >
        <Image
          bannerType="STAR"
          resizeMode="contain"
          source={1}
          style={
            Array [
              Object {
                "height": 125.44,
                "width": 686,
              },
            ]
          }
          width={686}
        />
        <Text
          alignCenter={false}
          allowFontScaling={false}
          bannerType="STAR"
          color="#000000"
          fontFamily="BOLD"
          fontSize={22}
          isUnderlined={false}
          isUppercase={true}
          maxFontSizeMultiplier={1.3}
          size="BANNER_LARGE"
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "CalibreBold",
                "fontSize": 34.5,
                "justifyContent": "center",
                "textAlign": "left",
                "textDecorationColor": "#000000",
                "textDecorationLine": "none",
                "textDecorationStyle": "solid",
                "textTransform": "uppercase",
              },
              Object {
                "color": "#FFFFFF",
                "letterSpacing": 1.5,
                "position": "absolute",
              },
            ]
          }
        >
          VIP
        </Text>
      </View>
      <View
        height={177}
        onLayout={[Function]}
        paddingTop={40}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderBottomLeftRadius": 16,
              "borderBottomRightRadius": 16,
              "height": 354,
              "justifyContent": "flex-start",
              "marginTop": -40,
              "paddingHorizontal": 42,
              "paddingTop": 80,
              "shadowColor": "rgba(53, 53, 53, 0.3)",
              "shadowOffset": Object {
                "height": 4,
                "width": 4,
              },
              "shadowOpacity": 1,
              "shadowRadius": 2,
              "width": "85%",
              "zIndex": -1,
            },
          ]
        }
      >
        <View />
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginHorizontal": 10,
              "marginVertical": 10,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(96,202,142,1)",
              "borderRadius": 88,
              "height": 176,
              "justifyContent": "center",
              "marginTop": -92,
              "opacity": 1,
              "width": 542,
            }
          }
          testID="dismiss-button"
        >
          <View
            borderRadius={44}
            colorScheme="rgba(128,229,173,1)"
            height={88}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(128,229,173,1)",
                  "borderRadius": 88,
                  "height": 166,
                  "justifyContent": "center",
                  "width": 536,
                },
              ]
            }
            width={271}
          >
            <View
              borderRadius={44}
              colorScheme="rgba(114,213,158,1)"
              height={88}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "rgba(114,213,158,1)",
                    "borderRadius": 108,
                    "height": 162,
                    "justifyContent": "center",
                    "marginTop": 8,
                    "width": 536,
                  },
                ]
              }
              width={271}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Text
                  alignCenter={false}
                  allowFontScaling={true}
                  color="#FFFFFF"
                  fontFamily="MEDIUM"
                  isUnderlined={false}
                  isUppercase={false}
                  maxFontSizeMultiplier={1.3}
                  size="XLARGE"
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontFamily": "Calibre-Medium",
                        "fontSize": 36,
                        "justifyContent": "center",
                        "textAlign": "left",
                        "textDecorationColor": "#FFFFFF",
                        "textDecorationLine": "none",
                        "textDecorationStyle": "solid",
                        "textTransform": "none",
                      },
                      Object {
                        "marginTop": 14,
                      },
                    ]
                  }
                >
                  Awesome!
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`Component VipModalWrapper renders correctly when showChicken 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={true}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={true}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "backgroundColor": "rgba(34,34,34, 0.6)",
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Array [
            Object {
              "height": 176,
              "marginBottom": 48,
              "width": 176,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
        width={686}
      >
        <Image
          bannerType="STAR"
          resizeMode="contain"
          source={1}
          style={
            Array [
              Object {
                "height": 125.44,
                "width": 686,
              },
            ]
          }
          width={686}
        />
        <Text
          alignCenter={false}
          allowFontScaling={false}
          bannerType="STAR"
          color="#000000"
          fontFamily="BOLD"
          fontSize={22}
          isUnderlined={false}
          isUppercase={true}
          maxFontSizeMultiplier={1.3}
          size="BANNER_LARGE"
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "CalibreBold",
                "fontSize": 34.5,
                "justifyContent": "center",
                "textAlign": "left",
                "textDecorationColor": "#000000",
                "textDecorationLine": "none",
                "textDecorationStyle": "solid",
                "textTransform": "uppercase",
              },
              Object {
                "color": "#FFFFFF",
                "letterSpacing": 1.5,
                "position": "absolute",
              },
            ]
          }
        >
          VIP
        </Text>
      </View>
      <View
        height={177}
        onLayout={[Function]}
        paddingTop={40}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderBottomLeftRadius": 16,
              "borderBottomRightRadius": 16,
              "height": 354,
              "justifyContent": "flex-start",
              "marginTop": -40,
              "paddingHorizontal": 42,
              "paddingTop": 80,
              "shadowColor": "rgba(53, 53, 53, 0.3)",
              "shadowOffset": Object {
                "height": 4,
                "width": 4,
              },
              "shadowOpacity": 1,
              "shadowRadius": 2,
              "width": "85%",
              "zIndex": -1,
            },
          ]
        }
      >
        <View />
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginHorizontal": 10,
              "marginVertical": 10,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(96,202,142,1)",
              "borderRadius": 88,
              "height": 176,
              "justifyContent": "center",
              "marginTop": -92,
              "opacity": 1,
              "width": 542,
            }
          }
          testID="dismiss-button"
        >
          <View
            borderRadius={44}
            colorScheme="rgba(128,229,173,1)"
            height={88}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(128,229,173,1)",
                  "borderRadius": 88,
                  "height": 166,
                  "justifyContent": "center",
                  "width": 536,
                },
              ]
            }
            width={271}
          >
            <View
              borderRadius={44}
              colorScheme="rgba(114,213,158,1)"
              height={88}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "rgba(114,213,158,1)",
                    "borderRadius": 108,
                    "height": 162,
                    "justifyContent": "center",
                    "marginTop": 8,
                    "width": 536,
                  },
                ]
              }
              width={271}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Text
                  alignCenter={false}
                  allowFontScaling={true}
                  color="#FFFFFF"
                  fontFamily="MEDIUM"
                  isUnderlined={false}
                  isUppercase={false}
                  maxFontSizeMultiplier={1.3}
                  size="XLARGE"
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontFamily": "Calibre-Medium",
                        "fontSize": 36,
                        "justifyContent": "center",
                        "textAlign": "left",
                        "textDecorationColor": "#FFFFFF",
                        "textDecorationLine": "none",
                        "textDecorationStyle": "solid",
                        "textTransform": "none",
                      },
                      Object {
                        "marginTop": 14,
                      },
                    ]
                  }
                >
                  Awesome!
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <Image
      resizeMode="contain"
      source={1}
      style={
        Array [
          Object {
            "position": "absolute",
            "right": 0,
            "top": -104,
          },
        ]
      }
      top={0}
    />
  </View>
</Modal>
`;
